app/src/main/java/com/android/voyageur/ui/overview/Overview.kt

```diff
@@ -171,57 +129,190 @@ fun OverviewScreen(
+  ) {
+    when {
+      trips.isEmpty() -> EmptyTripsMessage()
+      else ->
+          TripsList(
+              trips = trips,
+              searchQuery = searchQuery,
+              tripsViewModel = tripsViewModel,
+              navigationActions = navigationActions,
+              userViewModel = userViewModel)
+    }
+  }
+}
+
+/**
+ * Composable that displays a message when no trips are available.
+ *
+ * Shows a centered message prompting the user to create their first trip.
+ */
+@Composable
+private fun EmptyTripsMessage() {
+  Box(modifier = Modifier.fillMaxSize(), contentAlignment = Alignment.Center) {
+    Text(
+        modifier = Modifier.testTag("emptyTripPrompt"),
+        text = stringResource(R.string.empty_trip_prompt),
+    )
+  }
+}
+
+/**
+ * Composable that renders the list of trips.
+ *
+ * Displays a scrollable list of trip items, filtered according to the search query. Shows a "no
+ * results" message when the search yields no matches.
+ *
+ * @param trips List of all available trips
+ * @param searchQuery Current search query for filtering trips
+ * @param tripsViewModel ViewModel containing trips-related state and logic
+ * @param navigationActions Actions to handle navigation between screens
+ * @param userViewModel ViewModel containing user-related state and logic
+ */
+@Composable
+private fun TripsList(
+    trips: List<Trip>,
+    searchQuery: String,
+    tripsViewModel: TripsViewModel,
+    navigationActions: NavigationActions,
+    userViewModel: UserViewModel
+) {
+  val filteredTrips = filterTrips(trips, searchQuery)
+
+  if (searchQuery.isNotEmpty() && filteredTrips.isEmpty()) {
+    NoResultsFound(modifier = Modifier.testTag("noSearchResults"))
+    return
+  }
+
+  LazyColumn(
+      verticalArrangement = Arrangement.spacedBy(10.dp, Alignment.Top),
+      horizontalAlignment = Alignment.CenterHorizontally,
+      modifier = Modifier.fillMaxSize().testTag("lazyColumn")) {
+        filteredTrips.forEach { trip ->
+          item {
+            TripItem(
+                tripsViewModel = tripsViewModel,
+                trip = trip,
+                navigationActions = navigationActions,
+                userViewModel = userViewModel)
+            Spacer(modifier = Modifier.height(10.dp))
           }
         }
-      })
+      }
+}
+
+/**
+ * Filters and sorts the list of trips based on a search query.
+ *
+ * If no search query is provided, returns all trips sorted by start date. If a search query exists,
+ * returns trips whose names contain the query (case-insensitive), sorted by start date.
+ *
+ * @param trips List of trips to filter
+ * @param searchQuery Search query to filter trips by
+ * @return Filtered and sorted list of trips
+ */
+private fun filterTrips(trips: List<Trip>, searchQuery: String): List<Trip> {
+  return if (searchQuery.isEmpty()) {
+    trips.sortedBy { it.startDate }
+  } else {
+    trips.filter { it.name.contains(searchQuery, ignoreCase = true) }.sortedBy { it.startDate }
```

Comment:
Here I was wondering how we should treat the sorting. As we are doing it now, we show the oldest trips first, which might seem counterintuitive. Maybe we reverse the sorting so we have the trip that is most in the future first, so if you want to see old trips you scroll down, instead of having to scroll through a lot of old trips to reach your current trip. Wdyt?

Reply:
I totally agree with you - showing the trips in this order is counter intuitive. I'll modify the sorting as you suggested!